components/formats-bsd/test/loci/formats/utests/out/TiffWriterTest.java

```diff
@@ -89,6 +103,55 @@
     return new Object[][] {{null, pixelTypesOther}, {COMPRESSION_UNCOMPRESSED, pixelTypesOther}, {COMPRESSION_LZW, pixelTypesOther}, 
       {COMPRESSION_J2K, pixelTypesJ2K}, {COMPRESSION_J2K_LOSSY, pixelTypesOther}, {COMPRESSION_JPEG, pixelTypesJPEG}};
   }
+  
+  @DataProvider(name = "tiling")
+  public Object[][] createTiling() {
+    
+    int[] tileSize = {1, 32, 43, 64};
+    boolean[] booleanValue = {true, false};
+    int[] channelCount = {1, 3};
+    int[] seriesCount = {1, 5};
+    String[] compression = {COMPRESSION_UNCOMPRESSED, COMPRESSION_LZW, COMPRESSION_J2K, COMPRESSION_J2K_LOSSY, COMPRESSION_JPEG};
+    int compressionPixelTypeSize = (2 * pixelTypesOther.length) + pixelTypesOther.length - 1 + pixelTypesJ2K.length + 2;
+    int paramSize = tileSize.length * compressionPixelTypeSize * 4 * channelCount.length * seriesCount.length;
+    Object[][] data = new Object[paramSize][];
+    int index = 0;
+    for (int i = 0; i < tileSize.length; i++) {
```

Comment:
You could probably use range-based for all these loops, to make it a bit simpler?